pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -222,8 +220,15 @@ type Controller struct {
 	// 'MemoryPressure', 'OutOfDisk' and 'DiskPressure'.
 	taintNodeByCondition bool
 
-	nodeUpdateChannels []chan *v1.Node
-	nodeUpdateQueue    workqueue.Interface
+	// Channel that is used to pass updating Nodes to the background.
+	// This increases the throughput of taint updates by parallelization
+	// and not blocking on long operations (which shouldn't be done from
+	// event handlers anyway).
+	nodeUpdateChannel chan *v1.Node
```

Comment:
From cl description:

> As I think multiple channels with hashing is unnecessary complexity. Btw - I'm not sure how much this helps fixing that issue, but this combined with a fix to process only nodes needing to be processed should make the model ~similar to our cloud-cidr-allocator which we know works quite fast.

The crucial part of it is "combined with a fix to process only nodes needing to be processed".
We are not there yet. And without that, it would kill us.
So this should not be even reviewed before we get there.